util/dvsim/SimCfg.py

```diff
@@ -118,13 +78,13 @@ def __init__(self, flow_cfg_file, proj_root, args):
             self.cov = False
 
         # Set default sim modes for unpacking
-        if self.waves is True:
+        if self.waves:
             self.en_build_modes.append("waves")
-        if self.cov is True:
+        if self.cov:
             self.en_build_modes.append("cov")
-        if args.profile is not None:
+        if args.profile:
```

Comment:
This is not incorrect, but it does discard information. args.profile will always be `None`, `'time'` or `'mem'`. I think that using `None` for "not defined" is a good practice in Python, and catches all sorts of silly bugs to do with empty strings being falsy.

The changes to the surrounding lines also seem a bit dubious to me: the existing code makes it a bit clearer what types are expected; your change removes that in-code documentation. Is there any advantage to doing so? If not, I'd suggest reverting this hunk.

Reply:
I changed them all back.

Reply:
Thanks. I agree that the code does the same thing, but I think the "more typed" version is maybe a bit clearer to read.